tests/internal/tls/validFor_test.go

```diff
@@ -0,0 +1,162 @@
+	id := createmockIdentity([]string{validDNS}, []string{})
+
+	err := ziti_tls.ValidFor(id, invalidDNS+":"+validPort)
+	if err == nil {
+		t.Errorf("Expected error for invalid hostname, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+invalidDNS+"]. is valid for: ["+validDNS+"]", err.Error())
+}
+
+func TestValidFor_ValidIPv4(t *testing.T) {
+	id := createmockIdentity([]string{}, []string{validIP4})
+
+	err := ziti_tls.ValidFor(id, validIP4+":"+validPort)
+	if err != nil {
+		t.Errorf("Expected valid IP, got error: %v", err)
+	}
+}
+
+func TestValidFor_InvalidIPv4(t *testing.T) {
+	id := createmockIdentity([]string{}, []string{validIP4})
+
+	err := ziti_tls.ValidFor(id, invalidIP4+":"+validPort)
+	if err == nil {
+		t.Errorf("Expected error for invalid IP, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+invalidIP4+"]. is valid for: ["+validIP4+"]", err.Error())
+}
+
+func TestValidFor_ValidIPv6(t *testing.T) {
+	id := createmockIdentity([]string{}, []string{validIP6})
+
+	err := ziti_tls.ValidFor(id, "["+validIP6+"]:"+validPort)
+	if err != nil {
+		t.Errorf("Expected valid IPv6, got error: %v", err)
+	}
+}
+
+func TestValidFor_InvalidIPv6(t *testing.T) {
+	id := createmockIdentity([]string{}, []string{validIP6})
+
+	err := ziti_tls.ValidFor(id, "["+invalidIP6+"]:"+validPort)
+	if err == nil {
+		t.Errorf("Expected error for invalid IPv6, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+invalidIP6+"]. is valid for: ["+validIP6+"]", err.Error())
+}
+
+func TestValidFor_ValidMixed(t *testing.T) {
+	id := createmockIdentity([]string{validDNS}, []string{validIP4})
+
+	err1 := ziti_tls.ValidFor(id, validDNS+":"+validPort)
+	err2 := ziti_tls.ValidFor(id, validIP4+":"+validPort)
+
+	if err1 != nil {
+		t.Errorf("Expected valid hostname, got error: %v", err1)
+	}
+	if err2 != nil {
+		t.Errorf("Expected valid IP, got error: %v", err2)
+	}
+}
+
+func TestValidFor_InvalidMixed(t *testing.T) {
+	id := createmockIdentity([]string{validDNS}, []string{validIP4})
+
+	err1 := ziti_tls.ValidFor(id, invalidDNS+":"+validPort)
+	err2 := ziti_tls.ValidFor(id, invalidIP4+":"+validPort)
+
+	if err1 == nil {
+		t.Errorf("Expected error for invalid hostname, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+invalidDNS+"]. is valid for: ["+validIP4+", "+validDNS+"]", err1.Error())
+	if err2 == nil {
+		t.Errorf("Expected error for invalid IP, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+invalidIP4+"]. is valid for: ["+validIP4+", "+validDNS+"]", err2.Error())
+}
+
+func TestValidFor_NoCerts(t *testing.T) {
+	id := createmockIdentity([]string{}, []string{})
+
+	err := ziti_tls.ValidFor(id, validDNS+":"+validPort)
+	if err == nil {
+		t.Errorf("Expected error for no valid certs, got nil")
+	}
+	assert.Equal(t, "identity is not valid for provided host: ["+validDNS+"]. is valid for: []", err.Error())
```

Comment:
I am generally not a big fan of testing error strings. 

The alternative is to create a custom error type, implement the Error interface, write a test for the Error interface, and then inspect your returned error for the expected error type with specific fields inside of it.

Providing a thought, not asking for a change.